brave/src/main/java/brave/ScopedSpan.java

```diff
@@ -0,0 +1,49 @@
+package brave;
+
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+
+/**
+ * Used to model the latency of an operation.
+ *
+ * Here's a typical example of synchronous tracing from perspective of the scoped span:
+ * <pre>{@code
+ * // Note span methods chain. Explicitly start the span when ready.
+ * ScopedSpan span = tracer.startScopedSpan("encode");
+ * try {
+ *   return encoder.encode();
+ * } catch (RuntimeException | Error e) {
+ *   span.error(e); // Unless you handle exceptions, you might not know the operation failed!
+ *   throw e;
+ * } finally {
+ *   span.finish(); // finish - start = the duration of the operation in microseconds
+ * }
+ * }</pre>
+ *
+ * <p>Note: this type is only useful for local operations. For remote operations, use {@link Span}
+ */
+public abstract class ScopedSpan {
```

Comment:
The design in #676 implies that this would be marked as not thread safe. I would expect a comment to that affect here

Reply:
will do.. good catch

Reply:
done. thanks